cmd/kubetoken/validator.go

```diff
@@ -0,0 +1,429 @@
+	return &ctxValidator{
+		ctxName:           ctxName,
+		k8sClientProvider: k8sClientProvider,
+	}
+}
+
+type errOktetoContextNotFound struct {
+	ctxName string
+}
+
+func (e errOktetoContextNotFound) Error() string {
+	return fmt.Sprintf("context '%s' not found in the okteto context store", e.ctxName)
+}
+
+type errIsNotOktetoCtx struct {
+	ctxName string
+}
+
+func (e errIsNotOktetoCtx) Error() string {
+	return fmt.Sprintf("context '%s' is not an okteto context", e.ctxName)
+}
+
+func (v *ctxValidator) Validate(ctx context.Context) error {
+	result := make(chan error, 1)
+	go func() {
+		if v.ctxName == "" {
+			result <- errEmptyContext
+			return
+		}
+		if !isURL(v.ctxName) {
+			v.ctxName = okteto.K8sContextToOktetoUrl(ctx, v.ctxName, "", v.k8sClientProvider)
+		}
+		okCtx, exists := okteto.ContextStore().Contexts[v.ctxName]
+		if !exists {
+			result <- errOktetoContextNotFound{v.ctxName}
+			return
+		}
+		if !okCtx.IsOkteto {
+			result <- errIsNotOktetoCtx{v.ctxName}
+			return
+		}
+		result <- nil
+	}()
+
+	select {
+	case err := <-result:
+		return err
+	case <-ctx.Done():
+		return ctx.Err()
+	}
+}
+
+// nsValidator validates that the namespace is valid and accessible
+type nsValidator struct {
+	ns                   string
+	nsAccessChecker      nsAccessChecker
+	previewAccessChecker previewAccessChecker
+}
+
+type errNamespaceForbidden struct {
+	ns string
+}
+
+func (e errNamespaceForbidden) Error() string {
+	return fmt.Sprintf("you don't have access to the namespace '%s'", e.ns)
+}
+
+type errPreviewForbidden struct {
+	ns string
+}
+
+func (e errPreviewForbidden) Error() string {
+	return fmt.Sprintf("you don't have access to the preview '%s'", e.ns)
+}
+
+type nsAccessChecker struct {
+	oktetoClientProvider types.OktetoClientProvider
+}
+
+func newOktetoNsAccessChecker(oktetoClientProvider types.OktetoClientProvider) nsAccessChecker {
+	return nsAccessChecker{
+		oktetoClientProvider: oktetoClientProvider,
+	}
+}
+
+func (c *nsAccessChecker) hasAccess(ctx context.Context, ns string) (bool, error) {
+	okClient, err := c.oktetoClientProvider.Provide()
+	if err != nil {
+		return false, err
+	}
+	nList, err := okClient.Namespaces().List(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	for i := range nList {
+		if nList[i].ID == ns {
+			return true, nil
+		}
+	}
+	return false, errNamespaceForbidden{ns}
+}
+
+type previewAccessChecker struct {
+	oktetoClientProvider types.OktetoClientProvider
+}
+
+func newOktetoPreviewAccessChecker(oktetoClientProvider types.OktetoClientProvider) previewAccessChecker {
+	return previewAccessChecker{
+		oktetoClientProvider: oktetoClientProvider,
+	}
+}
+
+func (c *previewAccessChecker) hasAccess(ctx context.Context, preview string) (bool, error) {
+	okClient, err := c.oktetoClientProvider.Provide()
+	if err != nil {
+		return false, err
+	}
+	previewList, err := okClient.Previews().List(ctx, []string{})
```

Comment:
If we know the name of the preview I think we could try to get it instead list all of them